mobileChatExamples/androidChatExample/app/src/main/java/com/blitz/androidchatexample/viewmodel/ChatViewModel.kt

```diff
@@ -16,124 +17,213 @@ import com.amazonaws.services.connectparticipant.model.CreateParticipantConnecti
+        set(value) {
+            sharedPreferences.edit().putString("participantToken", value).apply()
+            _liveParticipantToken.value = value  // Reflect the new value in LiveData
+        }
+
+    fun clearContactId() {
+        sharedPreferences.edit().remove("contactID").apply()
+        _liveContactId.value = null
+    }
+
+    fun clearParticipantToken() {
+        sharedPreferences.edit().remove("participantToken").apply()
+        _liveParticipantToken.value = null
+    }
+
+    fun initiateChat() {
         viewModelScope.launch {
             _isLoading.value = true
-            // Clear the messages
-            _messages.postValue(emptyList())
-            try {
-                // Create StartChatRequest from configuration
-                val participantDetails =
-                    ParticipantDetails(displayName = chatConfiguration.customerName)
-                val request = StartChatRequest(
-                    connectInstanceId = chatConfiguration.connectInstanceId,
-                    contactFlowId = chatConfiguration.contactFlowId,
-                    participantDetails = participantDetails
-                )
-                val response = chatRepository.startChat(startChatRequest = request)
-                _startChatResponse.value = response
-                if (startChatResponse != null) {
-                    createParticipantConnection()
-                }else{
-                    _isLoading.postValue(false)
-                }
-            } catch (e: Exception) {
-                _startChatResponse.value = Resource.Error(e.localizedMessage ?: "Unknown Error")
-                _isLoading.postValue(false)
+            _messages.postValue(emptyList()) // Clear existing messages
+            if (participantToken != null) {
+                participantToken?.let { createParticipantConnection(it) }
+            } else if (contactId != null) {
+                startChat(contactId)
+            } else {
+                startChat(null) // Start a fresh chat if no tokens are present
             }
         }
     }
 
-    private fun createParticipantConnection() {
-        val startChatResponse = startChatResponse.value
-        if (startChatResponse is Resource.Success) {
-            val chatResponseData = startChatResponse.data?.data
-            // Now you can use chatResponseData
-            if (chatResponseData != null) {
-                chatRepository.createParticipantConnection(
-                    chatResponseData.startChatResult.participantToken,
-                    object :
-                        AsyncHandler<CreateParticipantConnectionRequest, CreateParticipantConnectionResult> {
-                        override fun onError(exception: Exception?) {
-                            // Handle error
-                            print(exception)
-                            _isLoading.postValue(false)
-                        }
+    private fun startChat(contactId: String?) {

```

Comment:
Do you think it's more readable if we call it `sourceContactId`, the same name in the API doc? It may not be obvious why we allow passing a contactId here and the API will return another contactId. It's fine if this is SDK but we provide this as an example which means customers may read and update some code here for their specific use case

Reply:
Yes that will make it clear. Thanks